modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/FieldInPageNavigationFormatter.php

```diff
@@ -0,0 +1,201 @@
+        'href' => "#" . $id,
+        'label' => $label,
+      ];
+      $element[$value]['#id'] = $id;
+      $element[$value]['#attributes']['id'] = $id;
+    }
+    $element['#links'] = $links;
+    if (count($links) > 0) {
+      $element['#inline_title'] = t('Page contents');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preRender(&$element, $rendering_object) {
+    parent::preRender($element, $rendering_object);
+    $this->process($element, $rendering_object);
+  }
+
+  /**
+   * Check if the field should create an in page navigation entry.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $field
+   *   The field to check.
+   * @param array $fieldgroups
+   *   List of groups.
+   *
+   * @return bool
+   *   Include or not the field.
+   */
+  private function excludeField(string $field_name, array $field, array $fieldgroups) :bool {
+    $type = $this->groupOrField($field_name, $fieldgroups);
+    switch ($type) {
+      case "field":
+        // Exclude empty fields.
+        if ($type === 'field' && !isset($field['#items'])) {
+          return TRUE;
+        }
+
+        break;
+
+      case 'group':
+        // Exclude empty groups.
+        if (count($field) < 1) {
+          return TRUE;
+        }
+        foreach ($field as $value) {
+          // If there is a field with values, don't exclude the group.
+          if (isset($value['#items'])) {
+            return FALSE;
+          }
+        }
+        // Exclude groups with only empty fields.
+        return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Get the label used on the field.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $processed_object
+   *   Entity processed.
+   *
+   * @return string
+   *   Label of the field.
+   */
+  private function fieldGetLabel(string $field_name, array $processed_object): string {
+    $label = '';
+    $type = $this->groupOrField($field_name, $processed_object['#fieldgroups']);
+    switch ($type) {
+      case "field":
+        $label = $this->getFieldLabel($processed_object[$field_name]);
+        break;
+
+      case 'group':
+        $label = $this->getGroupLabel($processed_object['#fieldgroups'][$field_name]);
+        break;
+    }
+    return $label;
+  }
+
+  /**
+   * Returns if the field is a group or a field.
+   *
+   * @param string $field_name
+   *   The name of the field.
+   * @param array $fieldgroups
+   *   List of groups.
+   *
+   * @return string
+   *   The type of field, group or field.
+   */
+  private function groupOrField(string $field_name, array $fieldgroups): string {
+    if (in_array($field_name, array_keys($fieldgroups))) {
+      return 'group';
+    }
+    return 'field';
+  }
+
+  /**
+   * Get the label of the field.
+   *
+   * @param array $field
+   *   Field info.
+   *
+   * @return string
+   *   The label of the field.
+   */
+  private function getFieldLabel(array $field): string {
```

Comment:
You are using this function only once, could we merge it with fieldGetLabel and rename that function to getFieldLabel?

Reply:
I merged getFieldLabel into fieldGetLabel and kept getGroupLabel as a separat method.